### PR TITLE
Feature/data attributes

### PR DESCRIPTION
**Pull request by lievenvg**
## PR Checklist

This PR fulfills the following requirements:

- [X] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] A changelog entry has been added to CHANGELOG.md if necessary

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [X] Feature
- [X] Code style update (formatting, local variables)
- [X] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] react application / infrastructure changes
- [ ] Other... Please describe:

Extension of Pull request #102
Added data-qa attributes and their corresponding unit tests to every component.

## What is the current behavior?

Until now only the Accordion had the possibility to have a data-qa attribute.

Issue Number: #100 

## What is the new behavior?
All components have the possibility to receive a data-qa attribute as a prop.

## Does this PR introduce a breaking change?

- [ ] Yes
- [X] No

## Resolved issues
Resolves: #100

### CODE REVIEW

File: packages/calendar/src/Calendar.js [hunk 1]
```diff
@@ -62,26 +64,27 @@ class Calendar extends Component<Prop> {
...
-			selectedDates,
-			minDate,
-			maxDate,
-			noWeekends,
+      format,
+      selectedDates,
+      minDate,
+      maxDate,
+      noWeekends,
+      qa,
```
Comment (TriangleJuice) [inline packages/calendar/src/Calendar.js]: Indenting

File: packages/calendar/src/Calendar.js [hunk 2]
```diff
@@ -62,26 +64,27 @@ class Calendar extends Component<Prop> {
...
+		return <div className="m-datepicker is-open" data-qa={qa}>
 			<DatePicker
 				format={format}
 				activeDate={activeDate}
 				selectedDates={selectedDates}
 				minDate={Moment(minDate, format)}
 				maxDate={Moment(maxDate, format)}
 				noWeekends={noWeekends}
-				clickOnDate={this.changeDate.bind(this)}
+        clickOnDate={this.changeDate.bind(this)}
```
Comment (TriangleJuice) [inline packages/calendar/src/Calendar.js]: Indenting

File: packages/form/src/Datepicker/Datepicker.js [hunk 3]
```diff
@@ -156,21 +158,22 @@ class Datepicker extends Component<Props> {
...
+      disabled,
+      readOnly,
+      name,
+      format,
+      selectedDates,
+      minDate,
+      maxDate,
+      noWeekends,
+      onBlur,
+      qa,
```
Comment (TriangleJuice) [inline packages/form/src/Datepicker/Datepicker.js]: Indenting

File: packages/form/src/Datepicker/Datepicker.spec.js [hunk 4]
```diff
@@ -23,4 +23,9 @@ describe('Datepicker', () => {
 		appInstance.toggleCalendar();
 		expect(datepicker.state('open')).toBe(true);
 	});
+
+	it('should set the data-qa attribute', () => {
+    const datepicker = shallow(<Datepicker qa="id-1234" />);
+    expect(datepicker.props()).toHaveProperty('data-qa', 'id-1234');
```
Comment (TriangleJuice) [inline packages/form/src/Datepicker/Datepicker.spec.js]: Indenting

File: packages/modal/src/Modal.js [hunk 5]
```diff
@@ -170,6 +173,8 @@ export default class Modal extends React.Component<Props, State> {
       triggerText,
       size,
 
+      qa,
+
```
Comment (TriangleJuice) [inline packages/modal/src/Modal.js]: Unnecessary space

File: packages/modal/src/Modal.js [hunk 6]
```diff
@@ -193,6 +198,8 @@ export default class Modal extends React.Component<Props, State> {
           className={classNames('m-modal', sizeClass, className)}
           overlayClassName={classNames('m-overlay', 'is-active', overlayClassName)}
 
+          data-qa={qa}
+
```
Comment (TriangleJuice) [inline packages/modal/src/Modal.js]: Unnecessary space

File: packages/calendar/src/Calendar.js (hunk 1 above)
Comment (lievenvg) [reply packages/calendar/src/Calendar.js]: Done.

File: packages/calendar/src/Calendar.js (hunk 2 above)
Comment (lievenvg) [reply packages/calendar/src/Calendar.js]: Done.

File: packages/form/src/Datepicker/Datepicker.js (hunk 3 above)
Comment (lievenvg) [reply packages/form/src/Datepicker/Datepicker.js]: Done.

File: packages/form/src/Datepicker/Datepicker.spec.js (hunk 4 above)
Comment (lievenvg) [reply packages/form/src/Datepicker/Datepicker.spec.js]: Done.

File: packages/modal/src/Modal.js (hunk 5 above)
Comment (lievenvg) [reply packages/modal/src/Modal.js]: Done.

File: packages/modal/src/Modal.js (hunk 6 above)
Comment (lievenvg) [reply packages/modal/src/Modal.js]: Done.

File: packages/toggle/src/Toggle.js [hunk 7]
```diff
@@ -69,11 +71,12 @@ class Toggle extends Component<Props> {
...
-	    checkedButtonClass,
-	    uncheckedLabel,
-	    uncheckedIcon,
-	    uncheckedButtonClass,
+      checkedIcon,
+      checkedButtonClass,
+      uncheckedLabel,
+      uncheckedIcon,
+      uncheckedButtonClass,
+      qa,
```
[the remark this answers is not in the document]
Comment (lievenvg) [reply packages/toggle/src/Toggle.js]: Deze lijkt me net een correctie van een foute indenting. ;-)

File: packages/subfooter/src/SubFooter.js [hunk 8]
```diff
@@ -1,21 +1,23 @@
...
 import StandardSubFooter from './Blocks/StandardSubFooter';
 
-import './Subfooter.scss';
+import './SubFooter.scss';
 
 type Props = {
   /** Get standard subfooter such as for type 'antwerpen' */
   type?: string,
   children?: any,
+  /** Qa id */
```
Comment (stefanpante) [inline packages/subfooter/src/SubFooter.js]: voor mij was het wel echt niet nodig da je die (niet veel zeggende) comment in elke file toevoegd :p (niet nodig om dit aan te passen)

File: packages/switch/src/Switch.js [hunk 9]
```diff
@@ -33,7 +33,8 @@ class Switch extends Component {
 			labelTrue,
 			labelFalse,
 			required,
-			disabled,
+      disabled,
+      qa,
```
Comment (stefanpante) [inline packages/switch/src/Switch.js]: klopt de indentation hier wel?

File: packages/subfooter/src/SubFooter.js (hunk 8 above)
Comment (lievenvg) [reply packages/subfooter/src/SubFooter.js]: Noted. 😉

File: packages/switch/src/Switch.js (hunk 9 above)
Comment (lievenvg) [reply packages/switch/src/Switch.js]: Nope. Bij deze aangepast.

**Pull request merged**